### ISSUE TITLE
`system_requirements()` doesn't run again after failure

### ISSUE THREAD

**Issue by rietmann** [conan-io/conan#838]
According to the documentation, `system_requirements()` tries to intelligently cache itself so that a tool like `apt-get` doesn't get run too often. However, if I have a bug in my `system_requirements()`, and it fails, it doesn't rerun according to this caching, even after a re-export of the package. It requires a full package deletion in order to purge the cache and trigger a rerun.

I don't know if it's possible, but only caching it upon a successful run would make more sense.

**Comment by memsharded** [comment]
Yes, totally makes sense. The current implementation uses:

```python
        try:
            output = conan_file.system_requirements()
        except Exception as e:
            coutput.error("while executing system_requirements(): %s" % str(e))
            raise ConanException("Error in system requirements")
```
I guess we should make helpers raise Exceptions if failing. What would be the expected behavior?

- If some system_requirements fail to install, raise and stop installation
- If some system_requirements fails, just output error, not log the state of installation as successful, but continue (which might fail later, at build or runtime, due to the missing system requirement)

**Comment by rietmann** [comment]
This seems like the expected behavior to me:
> If some system_requirements fail to install, raise and stop installation
At the least, this is what I want to happen.

**Comment by lasote** [comment]
Fixed in PR #901 Will be available in 0.19
